core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -243,18 +243,19 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
       applications.get(appId).flatMap { appInfo =>
         appInfo.attempts.find(_.attemptId == attemptId).flatMap { attempt =>
           val replayBus = new ReplayListenerBus()
+          val fileStatus = fs.getFileStatus(new Path(logDir, attempt.logPath))
+          val appListener = replay(fileStatus, isApplicationCompleted(fileStatus), replayBus)
+
           val ui = {
             val conf = this.conf.clone()
             val appSecManager = new SecurityManager(conf)
-            SparkUI.createHistoryUI(conf, replayBus, appSecManager, appInfo.name,
-              HistoryServer.getAttemptURI(appId, attempt.attemptId), attempt.startTime)
+            SparkUI.createHistoryUI(conf, replayBus, appSecManager,
+              appListener.appSparkVersion.getOrElse(""), appInfo.name,
+              HistoryServer.getAttemptURI(appId, attempt.attemptId),
+              attempt.startTime)
             // Do not call ui.bind() to avoid creating a new server for each application
           }
 
-          val fileStatus = fs.getFileStatus(new Path(logDir, attempt.logPath))
-
-          val appListener = replay(fileStatus, isApplicationCompleted(fileStatus), replayBus)
```

Comment:
Is it correct to remove this two lines? As I know it is necessary because in the previous `replay` call we only parse `SparkListenerApplicationStart` and `SparkListenerApplicationEnd`.